cfep-02.md

```diff
@@ -0,0 +1,85 @@
+
+<table>
+<tr><td> Title </td><td> Upgrade default macOS Travis image to `xcode6.4` </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Andreas Kloeckner &lt;inform@tiker.net&gt;</td></tr>
+<tr><td> Created </td><td> Nov 8, 2016</td></tr>
+<tr><td> Updated </td><td> Nov 8, 2016</td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda-forge/conda-forge-enhancement-proposals/pull/6 </td></tr>
+<tr><td> Implementation </td><td> (listed below) </td></tr>
+</table>
+
+## Abstract
+
+Upgrade the default Travis image to `xcode6.4`.
+
+## Implementation
+
+*   Adapt `conda smithy` so that it applies that edit on rerender.
+    [One-liner change here](https://github.com/conda-forge/conda-smithy/blob/b1d1730c4b8c8dc849464b1b39569b0e61ec3130/conda_smithy/templates/travis.yml.tmpl#L12)
+*   Apply the change in the `staged-recipes` master branch.
+    [One-liner change here](https://github.com/conda-forge/staged-recipes/blob/3ce42e083cebe5f39eea82d069d5a94cb7719218/.travis.yml#L5)
+
+    [PR](https://github.com/conda-forge/staged-recipes/pull/1094)
+*   Rebuild all packages with this? (TBD? Opinions?)
+
+    Likely not necessary given [C++ ABI Compatibility][cxxabi].
+*   Suggest that all new packages include the `toolchain` so that
```

Comment:
I still advocate the `toolchain` and it is versioned. We can pin the versions in the feedstocks. We can even add the pinned version to the pinning script. That all being said, I feel like Filipe and I have had this exact conversation over and over without a different result. So am not eager to dive into it again.

Instead I would like to find a reasonable middle ground that everyone can agree on and is easy to implement. As we do actually need to set the Mac OS X deployment target. My proposal would be to set the `MACOSX_DEPLOYMENT_TARGET` in `conda-forge-build-setup` as show in PR ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/46 ). @msarahan helped me with a PR to whitelist the `MACOSX_DEPLOYMENT_TARGET` in `conda-build`. So this will work with `conda-build` 2.0.12+.

I think this proposal is nice for a few reasons. First it sets the deployment target everywhere immediately. So we don't have to worry about people who are unaware of this change accidentally breaking compatibility. Second it consolidates the effort between upgrading to `conda-build` 2 and upgrading the Travis CI image. We need `conda-build` 2 to leverage this feature. So the two efforts should not be "competing" for time, but working together. Third it moves the discussion away from compiler preferences to maintaining compatibility of binaries. As compatibility is the real concern we are dealing with here, it makes sense to make that the focus and this seems to the easiest most straightforward way to do that.